dashboard/app/views/devise/registrations/edit.html.haml

```diff
@@ -149,6 +149,7 @@
       permission_status: @permission_status,
       user_email: current_user.hashed_email,
       in_section: JSON.dump(current_user.student? ? current_user.sections_as_student.present? : nil),
+      providers: JSON.dump(current_user.providers.filter_map(&:presence)),
```

Comment:
https://github.com/code-dot-org/code-dot-org/blob/adae83a09e532c6849c051a34caf5e2fb111d834/dashboard/app/models/user.rb#L2642-L2648

Reply:
I believe this means you will have a `'sponsered'` provider for teacher-made secret word / password accounts. That works to cover that case, too! nice!

Reply:
Wait, I'm wrong because a normal secret word/picture `sponsored` account would still be `migrated` but have an empty list of providers. Is it just a lack of providers that you look for in that case?

Reply:
`current_user.providers.append(current_user.sponsored? && 'sponsored').filter_map(&:presence),` 🤭

Reply:
Doesn't the absence of the providers mean the account is "sponsored" (word/picture)?